### ISSUE TITLE
False negative about the rule NP_STORE_INTO_NONNULL_FIELD

### ISSUE THREAD

**Issue by Belle-PL** [spotbugs/spotbugs#2118]
Hi, I found a false negative about the rule NP_STORE_INTO_NONNULL_FIELD, please read the following minimized code example. In line 11, `this.b` is assigned by a null object `this.a`, but `this.b` is annotated by `@NonNull`. Hence, I think this is a false negative. Thanks.

```java
import lombok.NonNull;
public class C {
    public Object a;
    @NonNull Object b;
    public void foo1() {
        this.a = new Object();
        this.foo2();
    }
    public void foo2() {
        this.a = null;
        this.b = this.a;  // should report a warning in this line
    }
}
```

SpotBugs Version: 4.7.1
